spec/controllers/file_controller_spec.rb

```diff
@@ -0,0 +1,516 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
```

Comment:
This is a great test setup :clap: I'll have to tweak this once we fold in the `send_file` code, but I already have a pattern in place for that. And these are an excellent set of comprehensive tests.